services/web/server/tests/unit/with_dbs/04/workspaces/test_workspaces__moving_folders_between_workspaces.py

```diff
@@ -0,0 +1,272 @@
+    assert len(data) == 0
+
+    url = client.app.router["list_projects"].url_for().with_query({"folder_id": "null"})
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 0
+
+    url = client.app.router["list_folders"].url_for().with_query({"folder_id": "null"})
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 1
+
+    url = (
+        client.app.router["list_folders"]
+        .url_for()
+        .with_query({"folder_id": f"{first_folder['folderId']}"})
+    )
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 1
+
+    return f"{second_folder['folderId']}"
+
+
+async def _move_folder_to_workspace_and_assert(
+    client: TestClient, folder_id: str, workspace_id: str
+):
+    assert client.app
+
+    # MOVE
+    base_url = client.app.router["replace_folder_workspace"].url_for(
+        folder_id=folder_id,
+        workspace_id=workspace_id,
+    )
+    resp = await client.put(f"{base_url}")
+    await assert_status(resp, NO_CONTENT)
+
+    # ASSERT
+    url = (
+        client.app.router["list_projects"]
+        .url_for()
+        .with_query(
+            {
+                "folder_id": folder_id,
+                "workspace_id": workspace_id,
+            }
+        )
+    )
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 2
+
+    url = (
+        client.app.router["list_folders"]
+        .url_for()
+        .with_query(
+            {
+                "folder_id": folder_id,
+                "workspace_id": workspace_id,
+            }
+        )
+    )
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    assert len(data) == 1
+
+
+@pytest.mark.parametrize("user_role,expected", [(UserRole.USER, status.HTTP_200_OK)])
+async def test_moving_between_private_and_shared_workspaces(
+    client: TestClient,
+    logged_user: UserInfoDict,
+    expected: HTTPStatus,
+    mock_catalog_api_get_services_for_user_in_product: MockerFixture,
+    fake_project: ProjectDict,
+    workspaces_clean_db: None,
+):
+    assert client.app
+    setup_db(client.app)
+
+    moving_folder_id = await _setup_test(client, logged_user, fake_project)
+
+    # We will test these scenarios of moving folders:
+    # 1. Private workspace -> Shared workspace
+    # 2. Shared workspace A -> Shared workspace B
+    # 3. Shared workspace A -> Shared workspace A (Corner case - This endpoint is not used like this)
+    # 4. Shared workspace -> Private workspace
+    # 5. Private workspace -> Private workspace (Corner case - This endpoint is not used like this)
+
+    # create a new workspace
+    url = client.app.router["create_workspace"].url_for()
+    resp = await client.post(
+        url.path,
```

Comment:
I started removing all this from the code since i noticed that it was easy to miss e.g. the queries etc
Just use `f"{url}"` everywhere